From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 18 Dec 2007 13:01:36 -0500
Subject: [x86] Add warning to nmi failure message
Message-id: 47680B00.8040703@redhat.com
O-Subject: Re: [RHEL5 PATCH]: Add "WARNING" to NMI failure message
Bugzilla: 401631

Backport of upstream commit a3aaabd6b402d8b0ede5aa4a040e9fdbbfdf9116

Add "WARNING" to NMI failure output so that our RHTS scripts can detect
NMI problems.

Successfully compiled and tested by me.

Resolves BZ 401631.

Acked-by: Pete Zaitcev <zaitcev@redhat.com>
Acked-by: Brian Maly <bmaly@redhat.com>

diff --git a/arch/i386/kernel/nmi.c b/arch/i386/kernel/nmi.c
index fad7ad2..e55226e 100644
--- a/arch/i386/kernel/nmi.c
+++ b/arch/i386/kernel/nmi.c
@@ -172,7 +172,8 @@ static int __init check_nmi_watchdog(void)
 #endif
 		if (nmi_count(cpu) - prev_nmi_count[cpu] <= 5) {
 			endflag = 1;
-			printk("CPU#%d: NMI appears to be stuck (%d->%d)!\n",
+			printk(KERN_WARNING "WARNING: CPU#%d: NMI "
+			       "appears to be stuck (%d->%d)!\n",
 				cpu,
 				prev_nmi_count[cpu],
 				nmi_count(cpu));
diff --git a/arch/x86_64/kernel/nmi.c b/arch/x86_64/kernel/nmi.c
index 1fc773f..9ee8c4f 100644
--- a/arch/x86_64/kernel/nmi.c
+++ b/arch/x86_64/kernel/nmi.c
@@ -183,7 +183,8 @@ int __init check_nmi_watchdog (void)
 	for_each_online_cpu(cpu) {
 		if (cpu_pda(cpu)->__nmi_count - counts[cpu] <= 5) {
 			endflag = 1;
-			printk("CPU#%d: NMI appears to be stuck (%d->%d)!\n",
+			printk(KERN_WARNING "WARNING: CPU#%d: NMI "
+			       "appears to be stuck (%d->%d)!\n",
 			       cpu,
 			       counts[cpu],
 			       cpu_pda(cpu)->__nmi_count);
